fix: trusted profile refactor

jor2 opened the pull request:
## Summary

Fixes #192 - Template assignment fails when account IDs are computed values.

**Changes:**
- Use index-based keys (`Account-0`, `Account-1`) instead of ID-based keys for for_each
- Enterprise data sources only fetched when using `["all"]` option
- Removed validation that blocked computed values

**Breaking Change:**
- List reordering with specific IDs will cause resource recreation
- Resource keys changed from `Account-{id}` to `Account-{index}`

**Migration Guide (Avoid Resource Recreation):**
To preserve existing resources, add moved blocks mapping old addresses to new ones.
```
moved {
  from = ibm_iam_trusted_profile_template_assignment.account_settings_template_assignment_instance["Account-123456"]
  to   = ibm_iam_trusted_profile_template_assignment.account_settings_template_assignment_instance["Account-0"]
}

moved {
  from = ibm_iam_trusted_profile_template_assignment.account_settings_template_assignment_instance["Account-789012"]
  to   = ibm_iam_trusted_profile_template_assignment.account_settings_template_assignment_instance["Account-1"]
}
```

## Test Plan
- Verified plan succeeds with computed account IDs from module outputs
- Verified `["all"]` option still works with enterprise accounts

Commit:
fix: trusted profile refactor

Commit:
fix: trusted profile refactor

Commit:
fix: trusted profile refactor

Commit:
fix: trusted profile refactor

Commit:
fix: trusted profile refactor

Commit:
fix: trusted profile refactor

Commit:
fix: trusted profile refactor

jor2 commented:
/run pipeline

Commit:
fix: trusted profile refactor

jor2 commented:
/run pipeline

Commit:
fix: trusted profile refactor

jor2 committed:
Merge branch 'main' into fix-profile

jor2 committed:
Merge branch 'main' into fix-profile

MatthewLemmond reviewed (changes requested):
Two small changes to the locals

Commit:
fix: use real ids

MatthewLemmond reviewed (dismissed):
LGTM

Commit:
fix: tests

Commit:
fix: add upgrade test + fix

Commit:
fix: pre-commit checks

jor2 committed:
Merge branch 'main' into fix-profile

jor2 committed:
Merge branch 'main' into fix-profile

jor2 committed:
Merge branch 'main' into fix-profile

Commit:
fix: use index-based keys for template assignments

Use index-based keys instead of ID-based keys when assigning templates
to specific accounts/groups. This fixes the for_each dependency issue
when account IDs are computed values (e.g., from module outputs).

- Data sources now only fetched when using ["all"] option
- Removes validation that prevented computed values
- List reordering with specific IDs will cause resource recreation

Commit:
chore: remove test files

mukulpalit-ibm committed:
Merge branch 'main' into fix-profile

mukulpalit-ibm commented:
/run pipeline

Khuzaima05 committed:
Merge branch 'main' into fix-profile

Khuzaima05 commented:
/run pipeline

vbontempi committed:
Merge branch 'main' into fix-profile

vbontempi commented:
could we document in the release notes a way to avoid the resources recreation through an example (i.e. using moved statement)?

Khuzaima05 committed:
Merge branch 'main' into fix-profile

Khuzaima05 commented:
/run pipeline

vbontempi reviewed (approved):
ok

terraform-ibm-modules-ops commented:
:tada: This PR is included in version 4.0.0 :tada:

The release is available on [GitHub release](https://github.com/terraform-ibm-modules/terraform-ibm-trusted-profile/releases/tag/v4.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
